### PR TITLE
Support Elasticsearch 8.x

### PR DESCRIPTION
**Pull request by lippertto**
This PR addresses issue #228

### CODE REVIEW

File: testcontainers/elasticsearch.py [hunk 1]
```diff
@@ -10,10 +10,50 @@
...
+        logging.warning("Could not determine major version from image name '%s'. Will use %s",
+                image_name, _FALLBACK_VERSION)
+        return _FALLBACK_VERSION
+    else:
+        return int(regex_match.group(1))
+
+
+def _environment_by_version(version: int) -> Dict[str, str]:
+    """Returns environment variables required for each major version to work."""
+    if version == 6:
```
Comment (tillahoffmann) [inline testcontainers/elasticsearch.py:44]: Thoughts on using `version <= 6` here for backwards compatibility?

**Review by tillahoffmann — COMMENTED** [review]
Looks good, one small question.

File: testcontainers/elasticsearch.py (hunk 1 above)
Comment (lippertto) [reply testcontainers/elasticsearch.py:44]: elastic.co supports only three major versions at a time, and version 5 has been eol for 2.5 years.

But I have no strong opinion here, and I understand that there will be people out there who still use version 5. (Although, I doubt that they will use testcontainers)

What I can do is to add a separate case for versions <= 5 . For these versions I would use the same setting as for version 6 and log a warning that these are neither supported by elastic.co nor this project. What do you think?

Comment (lippertto) [reply testcontainers/elasticsearch.py:44]: I have tested version `5.6.16`, and that runs fine with the environment for version 6

Comment (tillahoffmann) [reply testcontainers/elasticsearch.py:44]: Okay, let's leave it as-is.

**Comment by tillahoffmann** [comment]
Thanks for fixing this!

**Pull request merged**